ports/esp32/machine_pwm.c

```diff
@@ -95,16 +106,46 @@ STATIC int set_freq(int newval) {
     }
 
     // Configure the new resolution and frequency
-    timer_cfg.duty_resolution = res;
-    timer_cfg.freq_hz = newval;
-    if (ledc_timer_config(&timer_cfg) != ESP_OK) {
-        timer_cfg.duty_resolution = ores;
-        timer_cfg.freq_hz = oval;
+    timer->duty_resolution = res;
+    timer->freq_hz = newval;
+    if (ledc_timer_config(timer) != ESP_OK) {
+        timer->duty_resolution = ores;
+        timer->freq_hz = oval;
         return 0;
     }
     return 1;
 }
 
+STATIC int found_timer(int freq, bool same_freq_only) {
+    int free_timer_found = -1;
+    int timer;
+    // Find a free PWM Timer using the same freq
+    for (timer = 0; timer < LEDC_TIMER_MAX; ++timer) {
+        if (timers[timer].freq_hz == freq) {
+            // A timer already use the same freq. Use it now.
+            return timer;
+        }
+        if (!same_freq_only && (free_timer_found == -1) && (timers[timer].freq_hz == -1)) {
+            free_timer_found = timer;
+            // Continue to check if a channel with the same freq is in used.
+        }
+    }
+
+    return free_timer_found;
+}
+
+// return true if the timer is in use in addition to curr_channel
```

Comment:
Use a capital 'R' to match the other comments?
Could maybe also use some clarification, I'm not 100% sure what 'in use in addition to curr_channel' means